validate/validator.py

```diff
@@ -13,6 +13,13 @@
 
 printer = ""
 
+# Well known regex mapping.
+regex_map = {
+    0: "",  # UNKNOWN
+    1: r'^:?[0-9a-zA-Z!#$%&\'*+-.^_|~\x60]+$', # HTTP_HEADER_NAME
+    2: r'^[^\u0000-\u0008\u000A-\u001F\u007F]*$' #HTTP_HEADER_VALUE
```

Comment:
What do you think of using the constant ENUM_NAME values here instead of literal ints as keys?

Reply:
There's gotta be another way to do this, but it seemed like i had to dig into the enumdescriptor to get the name for the value. Inside the template though, I could reference the `StringRules.KnownRegex` type and access the name via a `Name` method, but couldn't modify `s` to have a `pattern`. Probably there was a way to do it via this second route?

Reply:
bump?

Reply:
To be addressed for #316